### PR TITLE
Prevent code from overflowing pre block

### LINKED ISSUE #29: Fix code samples with overflowing content
For instance, in analysis/introduction:

screenshot 2015-12-22 16 59 36

### PR DESCRIPTION
**Pull request by ozan**
Quick review anybody?

Fixes #29 

Code is prevented form overflowing out of pre block, by a combination of using the more compact of the two monospace fonts more consistently; adding linebreaks; and, `overflow: scroll`.

### CODE REVIEW

File: book/graphs/implementation.md [hunk 1]
```diff
@@ -31,7 +31,10 @@ class Vertex(object):
         self.neighbors[neighbor] = weight
 
     def __str__(self):
-        return '{} neighbors: {}'.format(self.key, [x.key for x in self.neighbors])
+        return '{} neighbors: {}'.format(
```
Comment (smart-alek) [inline book/graphs/implementation.md:34]: Did this change actually affect spillover in those boxes? Or does this address character limits in a text editor?

**Comment by smart-alek** [comment]
A few things here:
1. Looks like `overflow-x: scroll` created a second line at the bottom of each pre block (see below). When I turned off this rule in the Chrome console, I still wasn't seeing any spillover, so maybe it can be safely removed?
2. Double check me on this, but the other monospaced font seems a _tad_ large for surrounding text when **not** in a pre block. Whenever a method name is mentioned in prose, it really looks like it's demanding attention. Can we distinguish between that font for its separate use cases?
3. There were a few changes you had to make (like `output_template`) that are purely cosmetic. While we want things to look good, would it be easier to increase the width of these pre blocks?

DOUBLE LINES
screen shot 2016-01-05 at 9 00 45 pm

BIG MONOSPACED FONT
screen shot 2016-01-05 at 9 01 23 pm

File: book/chapter.css [hunk 2]
```diff
@@ -117,13 +117,14 @@ ol.chapters {
...
 
-code[class*="language-"] {
+code, code[class*="language-"] {
   font-family: Inconsolata, Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
 }
 
 pre {
   background: rgba(242, 242, 242, 0.5);
   padding: 20px;
+  overflow-x: scroll;
```
Comment (smart-alek) [inline book/chapter.css]: Mentioned it in my general comment, but I think this rule created an extra line at the bottom of each pre box! Catastrophe!

**Comment by ozan** [comment]
Hmm, regarding the big monospaced font, this is what I see:

screen shot 2016-01-06 at 5 09 53 pm

That's both locally and on https://algos-book-pr-34.herokuapp.com/analysis/introduction/

Is the Inconsolata font loading for you? And you're definitely on this branch and did a hard refresh?

**Comment by ozan** [comment]
Re the "extra line" that looks like an OSX scrollbar... I should have `overflow: auto` rather than `overflow: scroll`, but I didn't notice since I have OSX configured to only show scrollbars when scrolling. Will fix.

**Comment by ozan** [comment]
Double line issue should be fixed in 97be8f9. Can you take another quick look, and also check that Inconsolata is loading on your set up? Should be good to merge after that.

File: book/graphs/implementation.md (hunk 1 above)
Comment (ozan) [inline book/graphs/implementation.md:34]: This must have been spilling over... the character limits I'm only enforcing in .py files as that's where my linter runs :smile:

**Comment by ozan** [comment]
Also re:

> While we want things to look good, would it be easier to increase the width of these pre blocks?

I think it's also better code :smile: . In general sticking to the 80-ish char guideline in Python tends to be a good idea.

**Comment by smart-alek** [comment]
Double line issue seems to be fixed, but Inconsolata still doesn't seem to be loading. Git bisect pointed me to [this commit](https://github.com/Bradfield/algos/commit/6edb768ecdfba4258a6fc3432992d5784f689da6). How else are we loading the Inconsolata font if not in that spot? Previous versions on this branch that look like your screenshots are defaulting to just monospace on my set up. :mag:

**Comment by ozan** [comment]
@aleksharma12 thanks! I've created #38 to track the font issue directly; will merge this in for now.

**Pull request merged**